app/services/messages/reader.rb

```diff
@@ -1,18 +1,21 @@
 module Messages
   class Reader
-    def initialize(message)
+    def initialize(message, visibility_time = nil, resolver_name = nil)
       @message = message
+      @visibility_time = visibility_time
+      @resolver_name = resolver_name
     end
```

Comment:
Also following the event approach, these attributes won't be needed here, and they don't really belong here because they are discord specifics

```suggestion
    def initialize(message)
      @message = message
    end
```

Reply:
Yeah, I moved the params of `#read_message`; the method can now receive the `payload`, which will be passed to the broadcasting